### PR TITLE
Add a possibility to send emails about crashes

### PR DESCRIPTION
**Pull request by pnuu**
As Trollflow2 handles crashes internally, `supervisord` don't report anything about them, and only trace of the problem is in the log files. This PR adds a possibility to send emails with the error message.

### CODE REVIEW

**Comment by pnuu** [comment]
The functionality is now more generic and user can supply their own crash handler.

**Comment by pnuu** [comment]
I think I'll still add a possibility to collect info from `input_mda` to be passed to the handlers.

File: trollflow2/__init__.py [hunk 1]
```diff
@@ -128,8 +128,11 @@ def save_datasets(job):
         fname_pattern = fmat['fname_pattern']
         filename = compose(os.path.join(fmat['output_dir'], fname_pattern), fmat)
         fmat.pop('format', None)
+        fmat.pop('filename', None)
```
Comment (mraspaud) [inline trollflow2/__init__.py]: Shouldn't it raise an error instead ? there shouldn't be a 'filename' there, right ?

Comment (pnuu) [reply trollflow2/__init__.py]: Yeah, I was wondering how the filename ends up in the format, but thought it was by design as it was there always for the 2nd and later composites. It is added in the `else` branch, and as I don't understand what `plist_iter` is doing, thought it somehow propagated to the later products.

Comment (mraspaud) [reply trollflow2/__init__.py]: I'm afraid that could be a sign that the format section is shared between multiple products.

**Pull request merged**